### ISSUE TITLE
autoprefixer-rails doesn't work with sprockets 3.x in Rails development environment

### ISSUE THREAD

**Issue by edariedl** [ai/autoprefixer-rails#102]
Hi, autoprefixer-rails does not process css during request in development envirenomnt. It worked with Sprockets `2.x` until last night when we upgraded Sprockets to `3.x`.

To be sure it is not issue with our application I've created new Rails 5 application with few controllers according to http://guides.rubyonrails.org/getting_started.html and added `autoprefixer-rails` gem.

My example `scss` code:

```scss
.form-row__inner {
  display: block;
  display: flex;
  align-items: baseline;
}
```

This was generated during request.

```css
.form-row__inner {
    display: block;
    display: flex;
    align-items: baseline;
}
```

Autoprefixer was not executed.

But `rake assets:precompile` works and creates prefixes in css file:

```css
.form-row__inner {
  display: block;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: baseline;
      -ms-flex-align: baseline;
          align-items: baseline;
}
```

**Comment by Sathors** [comment]
I can confirm the problem, using sprockets 3.6.1 on rails 4 in development.

When compiling the assets, the output css is vendor-prefixed all right.

**Comment by ai** [comment]
Main advice: go to Gulp/webpack. Sprockets is dead and I didn’t support this gem very well.

**Comment by ai** [comment]
Let’s try to debug code by this steps:

https://github.com/ai/autoprefixer-rails/issues/101#issuecomment-237838903

**Comment by ai** [comment]
@abeMedia is it connected with your warning fix?

**Comment by ai** [comment]
I think issue is in this commit https://github.com/ai/autoprefixer-rails/commit/cd9b2ef1d5a89042e4d97614490ab03fa895a653

But I didn’t know how to fix and not get warnings again (see #98)

**Comment by ai** [comment]
Seems like issue is for Sprockets < 3.7

**Comment by ai** [comment]
I try to fix it in Sprockets 3.6 ba46a8683786087d9215f45d742e93f399a4f77c

Could somebody test it from GitHub? Just replace `gem "autoprefixer-rails"` to `gem "autoprefixer-rails", github: "ai/autoprefixer-rails"`

**Comment by edariedl** [comment]
Wow, you are fast, thanks. Actually I use it with Sprockets 3.7 and it doesn't work.

**Comment by edariedl** [comment]
@ai After your fix, it works for Sprockets 3.6. Sprockets 3.7 is still broken.

**Comment by edariedl** [comment]
@ai It also works for Sprockets 4.0.0.beta2

**Comment by Sathors** [comment]
I can confirm it is working with Sprockets 3.6.

**Comment by ai** [comment]
:). Does somebody have idea, how to fix it in 3.7 and don't throw warning?

**Comment by ai** [comment]
Guys, please test @edariedl fix again by `gem "autoprefixer-rails", github: "ai/autoprefixer-rails"`

**Comment by ai** [comment]
I tested in our project — fix works with Sprockets 2.

**Comment by edariedl** [comment]
It works in our poject with Sprockets 3.7 too.

**Comment by ai** [comment]
@edariedl awesome. I am releasing your fix in 6.4.0.1.

**Comment by edariedl** [comment]
Thank you :)

**Comment by ai** [comment]
Released in 6.4.0.1

**Comment by abeMedia** [comment]
@ai my bad I had only tested with 3.7
